virttest/vt_utilities/cpu.py

```diff
@@ -0,0 +1,174 @@
+                "Cascadelake-Server-noTSX": ("avx512f,avx512dq,avx512bw,"
+                                             "avx512cd,avx512vl,clflushopt,"
+                                             "avx512_vnni"),
+                "Skylake-Server": "avx512f,clwb,xgetbv1,pcid,hle,rtm",
+                "Skylake-Server-noTSX-IBRS": "avx512f,clwb,xgetbv1,pcid",
+                "Skylake-Client": "xgetbv1,pcid,hle,rtm",
+                "Skylake-Client-noTSX-IBRS": "xgetbv1,pcid",
+                "Broadwell": "adx,rdseed,3dnowprefetch,hle,rtm",
+                "Broadwell-noTSX": "adx,rdseed,3dnowprefetch",
+                "Haswell": "fma,avx2,movbe,hle,rtm",
+                "Haswell-noTSX": "fma,avx2,movbe",
+                "IvyBridge": "f16c,fsgsbase,erms",
+                "SandyBridge": ("avx,xsave,aes,sse4_2|sse4.2,sse4.1|sse4_1,"
+                                "cx16,ssse3"),
+                "Westmere": "aes,sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                "Nehalem": "sse4.2|sse4_2,sse4.1|sse4_1,cx16,ssse3",
+                "Penryn": "sse4.1|sse4_1,cx16,ssse3",
+                "Conroe": "ssse3"}
+
+
+def get_cpu_info():
+    """
+    Return information about the CPU architecture
+
+    :return: cpu information in dict type
+    """
+    cmd = "lscpu | tee"
+    output = process.run(cmd, shell=True,
+                         ignore_status=True).stdout_text.splitlines()
+    return dict(map(lambda x: [i.strip() for i in x.split(":", 1)], output))
+
+
+def get_cpu_models():
+    """
+    Get cpu model from host cpuinfo.
+    Note: Do not use this function on the ppc64/ppc4le platform.
+    """
+    def _cpu_flags_sort(cpu_flags):
+        """
+        Update the cpu flags get from host to a certain order and format
+        """
+        flag_list = sorted(re.split("\s+", cpu_flags.strip()))
+        cpu_flags = " ".join(flag_list)
+        return cpu_flags
+
+    def _make_up_pattern(flags):
+        """
+        Update the check pattern to a certain order and format
+        """
+        pattern_list = sorted(re.split(",", flags.strip()))
+        pattern = r"(\b%s\b)" % pattern_list[0]
+        for i in pattern_list[1:]:
+            pattern += r".+(\b%s\b)" % i
+        return pattern
+
+    with open("/proc/cpuinfo") as fd:
+        cpu_info = fd.read()
+
+    cpu_flags = " ".join(get_cpu_flags(cpu_info))
+    vendor = get_cpu_vendor(cpu_info)
+
+    cpu_model = None
+    cpu_support_model = []
+    if cpu_flags:
+        cpu_flags = _cpu_flags_sort(cpu_flags)
+        for cpu_type in CPU_TYPES.get(vendor):
+            pattern = _make_up_pattern(CPU_TYPES_RE.get(cpu_type))
+            if re.findall(pattern, cpu_flags):
+                cpu_model = cpu_type
+                cpu_support_model.append(cpu_model)
+
+    return cpu_support_model
+
+
+def get_cpu_models_on_ppc64():
+    """
+    Get cpu model from ppc64/ppc64le cpuinfo
+    """
+    return []  # if platform is the ppc64/ppc64le
```

Comment:
Hi @maxujun 
Does the power platform support get the CPU mode on the host?
Thanks.

Reply:
> Hi @maxujun Does the power platform support get the CPU mode on the host? Thanks.

Hello @maxujun , can you give some comments?
Thanks!

Reply:
Yes, support.

Reply:
hello @zhencliu @luckyh @yanan-fu ,
do you guys have any comments? Thanks!

Reply:
Not support get cpu backward-compatibility mode list from host cpu info for power.only can get host self cpu model.